Refactor launch app AWS integration

amirfefer opened the pull request:
### Description
<!-- Must include 2-3 sentence summary of proposed changes -->
<!-- Must include links to impacted UI(s) or information regarding the impacted UI -->
<!-- Must include any relevant steps to reproduce (if not clear in tracked issue or story) -->
<!-- Must include RHCLOUDXXXX link (if proposed change involves tracked issue or story) -->

Currently, our AWS integration wizard guides users through creating a policy and role resources, attaching them manually via the AWS CLI, and entering the ARN.

By leveraging an AWS CloudFormation stack template, we can streamline this process. Users will be able to achieve this with a simple redirection to AWS console with a dedicated template, eliminating the need for CLI commands or copy any JSON file. The ARN will be retrieved automatically, simplifying the integration with our service.


[HMS-4041](https://issues.redhat.com/browse/HMS-4041)

---

### Screenshots
<!-- Before and after proposed changes is ideal -->
<!-- Any key UI permutations should be captured -->
<!-- Draw attention to the area of UI that has changed -->
#### Before:

https://github.com/RedHatInsights/sources-ui/assets/11807069/c8589562-2dd4-4b12-a2be-8198f97bee66



#### After:

https://github.com/RedHatInsights/sources-ui/assets/11807069/7b0b2ce4-6c55-486c-b9d8-efd15ca1d069



---

### Checklist ☑️
- [x] PR only fixes one issue or story <!-- open new PR for others -->
- [ ] Change reviewed for extraneous code <!-- console statements, comments, files, incorrect file renaming (not using `git mv`), whitespace, etc. -->
- [ ] UI best practices adhered to <!-- TODO: add a link; responsiveness, input sanitization, prioritizing PatternFly and FEC, feature gating, etc. -->
- [ ] Commits squashed and meaningfully named <!-- (2-3 commits per PR maximum, 1 is ideal) -->
- [ ] All PR checks pass locally (build, lint, test, E2E)

##
- [ ] _(Optional) QE: Needs QE attention (OUIA changed, perceived impact to tests, no test coverage)_
- [ ] _(Optional) QE: Has been mentioned_
- [ ] _(Optional) UX: Needs UX attention (end user UX modified, missing designs)_
- [x] _(Optional) UX: Has been mentioned_
##

fhlavac commented:
@amirfefer it looks like the tests need to be updated. Also, did anyone from UX review the changes?

I noticed especially the button not being in sync with what we have for Azure:
![Snímek obrazovky 2024-06-28 v 12 00 51](https://github.com/RedHatInsights/sources-ui/assets/50696716/d73b7358-1ed5-4f5e-bb03-37bbbe42d15d)
vs
![Snímek obrazovky 2024-06-28 v 12 02 29](https://github.com/RedHatInsights/sources-ui/assets/50696716/57548b63-2ec3-4ebc-87b2-84e39974d814)

And also the tooltip looks a bit weird
![Snímek obrazovky 2024-06-28 v 12 01 35](https://github.com/RedHatInsights/sources-ui/assets/50696716/03a4e564-647b-4016-94dd-1ada2a4c0657)

fhlavac commented on src/components/addSourceWizard/hardcodedComponents/aws/provisioningArn.js:
```diff
@@ -143,85 +95,129 @@ export const IAMRoleDescription = () => {
+        </StackItem>
+        <StackItem>
+          <Bullseye>
+            <Button isDisabled={!provAwsAccount} isLoading={!provAwsAccount} onClick={openAWSStackPage} variant="primary">
```
Here we could directly use 
```
<Button
        component="a"
        target="_blank"
        href={link}
        ...
```

maryshak1996 commented:
Hey there @fhlavac asked me to do a quick UX review here -- overall looks good, but I've got a few suggestions:

- We don't have to capitalize the 'r' in 'roles' every time we say 'IAM Role' -- we can just say 'IAM role'
- For actions that a user clicks on we should use **'[words]'** as well as bold the text. For example, instead of "Click on **Connect AWS** to open AWS...", we should say "Click on **'Connect AWS'** to open AWS..."
- Make the button will width to match the lighthouse example
- Instead of having a very large tooltip that expands beyond the size of the modal itself, we could do an expandable section that says 'See role permissions'. It would be closed by default (like you see in the first screenshot below) but then could be triggered open if the user would like (second screenshot).
![image](https://github.com/RedHatInsights/sources-ui/assets/19824574/282be70b-8283-4b7f-9ff7-60cbd4a367c7)
![image](https://github.com/RedHatInsights/sources-ui/assets/19824574/a8317994-0e5b-4332-bba5-286ceff7212a)

Hope this helps!

amirfefer commented:
Thanks @fhlavac and @maryshak1996!
I've addressed your comments

![Screenshot 2024-07-02 at 21 02 33](https://github.com/RedHatInsights/sources-ui/assets/11807069/b5215fc7-d16f-4318-9348-36b995df4a0d)

fhlavac commented:
/retest

fhlavac commented:
This should fix the build, just we need to bump the FEC version when it is released https://github.com/RedHatInsights/frontend-components/pull/2028

fhlavac commented:
[@redhat-cloud-services/tsc-transform-imports/v/1.0.13](https://www.npmjs.com/package/@redhat-cloud-services/tsc-transform-imports/v/1.0.13)

ezr-ondrej commented:
@fhlavac should we bump the version in this PR?

fhlavac commented:
@ezr-ondrej yes, please

ezr-ondrej commented:
/retest

ezr-ondrej commented:
outch: `Error: building at STEP "COPY LICENSE /licenses/": checking on sources under "/var/lib/jenkins/workspace/RedHatInsights-sources-ui-pr-check/build/container_workspace": copier: stat: "/LICENSE": no such file or directory`

ezr-ondrej commented:
Trying to fix it here: https://github.com/RedHatInsights/sources-ui/pull/1216

ezr-ondrej commented:
@RedHatInsights/platform-experience this would be amazing to have if you get to it :pray:
